src/v/config/configuration.cc

```diff
@@ -1282,6 +1282,65 @@ configuration::configuration()
        .example = "false",
        .visibility = visibility::user},
       true)
+  , audit_enabled(
+      *this,
+      "audit_enabled",
+      "Enable/Disable audit logging.",
+      {.needs_restart = needs_restart::no, .visibility = visibility::user},
+      false)
+  , audit_log_num_partitions(
+      *this,
+      "audit_log_num_partitions",
+      "Number of partitions for the internal audit log topic. Attempt to "
+      "create topic is only performed if it doesn't already exist, disable and "
+      "re-enable auditing for changes to take affect",
+      {.needs_restart = needs_restart::no, .visibility = visibility::user},
+      12)
+  , audit_log_replication_factor(
+      *this,
+      "audit_log_replication_factor",
+      "Replication factor of the internal audit log topic. Attempt to create "
+      "topic is only performed if it doesn't already exist, disable and "
+      "re-enable auditing for changes to take affect",
+      {.needs_restart = needs_restart::no, .visibility = visibility::user},
+      3)
+  , audit_client_max_buffer_size(
+      *this,
+      "audit_client_max_buffer_size",
+      "Maximum number of bytes the internal audit client will allocate for "
+      "audit log records. Disable and re-enable auditing for changes to take "
+      "affect",
+      {.needs_restart = needs_restart::no, .visibility = visibility::user},
+      1_MiB)
+  , audit_queue_drain_interval_ms(
+      *this,
+      "audit_queue_drain_interval_ms",
+      "Frequency in which per shard audit logs are batched to client for write "
+      "to audit log. Longer intervals allow for greater change for coalescing "
+      "duplicates (great for high throughput auditing scenarios) but increase "
+      "the risk of data loss during hard shutdowns.",
+      {.needs_restart = needs_restart::no, .visibility = visibility::tunable},
+      500ms)
+  , audit_max_queue_elements_per_shard(
+      *this,
+      "audit_max_queue_elements_per_shard",
+      "Maximum number of allowed elements in the audit buffers, per shard. "
+      "Once this value is reached, any request handlers that cannot enqueue "
+      "audit messages will return a non retryable error to the client. Note "
+      "that this only will occur when handling requests that are currently "
+      "enabled for auditing.",
+      {.needs_restart = needs_restart::no, .visibility = visibility::tunable},
+      100000)
```

Comment:
Do we have a ticket tracking the discussion about this metric's units vs an equivalent one based on memory size?

Reply:
https://github.com/redpanda-data/redpanda/issues/14099 i also added it to the audit logging epic

Reply:
Thanks

Reply:
@graphcareful - shouldn't this be a semaphore per shard, and the future of producing to an audit log event simply block the producer-future. Doesn't make sense to drop audit events, as this is the last defense for ppl debugging.

Reply:
Yup thats actually how it does work now, we will not drop audit events if the queue is full but rather block. #14099 just changes the metric by which we will do this, instead of number of elements (which doesn't really make sense to the end user anyway) to size in bytes per shard.